### ISSUE TITLE
Deployment script doesn't recognize `--network` flag; Kovan deployment doesn't work

### ISSUE THREAD

**Issue by tinybike** [AugurProject/turbo#13]
Short version: the custom deployment scripts use an `--environment` instead of `--network` parameter, which puts them in disharmony with the readme as well as hardhat.  Kovan deployment also isn't working.

Long version:

The readme says to run this command to deploy to the local hardhat chain:
```
yarn smart contracts:deploy --network localhost
```
However, the `--network` flag isn't recognized.  Instead, it is expecting an [`--environment` flag](https://github.com/AugurProject/turbo/blob/master/packages/smart/scripts/deploy-turbo.ts#L20), with argument [`local`](https://github.com/AugurProject/turbo/blob/master/packages/smart/src/environments/index.ts#L6-L9) instead of `localhost`.  When I run `yarn smart contracts:deploy --environment local`, it deploys successfully.

However, I can't get the contracts deployed to Kovan at all.  I tried the following (again, it expects `environment` instead of `network`):
```bash
jack@substrate:~/src/turbo$ PRIVATE_KEY=<0xwhatever> yarn smart contracts:deploy --environment kovan
yarn run v1.22.5
$ yarn workspace @turbo/smart contracts:deploy --environment kovan
$ ts-node --files ./scripts/deploy-turbo.ts --environment kovan
Error: unknown account #0 (operation="getAddress", code=UNSUPPORTED_OPERATION, version=providers/5.0.24)
    at Logger.makeError (/home/jack/src/turbo/node_modules/@ethersproject/logger/src.ts/index.ts:205:28)
    at Logger.throwError (/home/jack/src/turbo/node_modules/@ethersproject/logger/src.ts/index.ts:217:20)
    at /home/jack/src/turbo/node_modules/@ethersproject/providers/src.ts/json-rpc-provider.ts:149:24
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at Function.create (/home/jack/src/turbo/node_modules/@nomiclabs/hardhat-ethers/src/signers.ts:5:34)
    at main (/home/jack/src/turbo/packages/smart/scripts/deploy-turbo.ts:36:18) {
  reason: 'unknown account #0',
  code: 'UNSUPPORTED_OPERATION',
  operation: 'getAddress'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 1
Command: /home/jack/.nvm/versions/node/v14.16.0/bin/node
Arguments: /usr/share/yarn/lib/cli.js contracts:deploy --environment kovan
Directory: /home/jack/src/turbo/packages/smart
Output:
```
It's not clear to me if the Kovan deployment issue is caused by the flag confusion, or is something different.  (Or, maybe I'm just messing something up...?)

I have a small suggestion and a larger one.  The small suggestion is just to make sure that your readme is harmonized with your code.  The larger suggestion (from the perspective of someone who knows the pain of maintaining custom deployment machinery), is to avoid maintaining custom deployment machinery, if it's at all possible.  For example, if there's not a compelling reason to use `--environment` over `--network` (and `local` over `localhost`), use `--network` and `localhost` so that the flags match the underlying hardhat flags.  Similarly, is there anything in `src/environments` that couldn't just be specified in the hardhat config file?  (I see the kovan endpoint hardcoded both in the config file and in `src/environments/kovan.ts`...)

**Comment by robert-e-davidson3** [comment]
Ah you're right I changed the config mechanism and forgot to update the readme for it. Sorry :/

I wrote the deploy script before I knew much about hardhat tasks. It will be migrated into a task which might clean up the configuration strategy. But if that doesn't, I don't want to fragment the config. I know hardhat plugins can alter the config even if I can't figure out where the type information is getting added so that might be the strategy going forward: make hardhat config large enough to act as the general Turbo config.

The broken kovan deploy is from the config change. Hardhat provides a provider with a signer but I need to do that manually now. I'll fix this for now (PR later today) but the task task will change that mechanism anyway.

Thanks for the feedback!
